sound/soc/sof/loader.c

```diff
@@ -854,6 +854,10 @@ EXPORT_SYMBOL(snd_sof_run_firmware);
 
 void snd_sof_fw_unload(struct snd_sof_dev *sdev)
 {
-	/* TODO: support module unloading at runtime */
+	/* Don't cache a firmware that failed to boot.
+	 * TODO: fully support module unloading at runtime too.
+	 */
```

Comment:
we don't have such plans at the moment, let's remove this to avoid issues when upstreaming.

Reply:
I'm confused by this existing TODO, should I leave it as is? Even though I'm adding code below it? Or you asking to remove the comment? But you wrote "we don't have such plans".

I'm confused because we're testing this TODO regularly and it is what this PR fixes.

Reply:
By editing an existing comment, you make it unclear what exactly Intel will do.
This patch only releases firmware on errors. You can either 
a) remove the TODO in a separate patch, or
b) keep it as is for now.

My point was: don't edit something that you didn't modify, and the 'fully' statement is not useful.